tests/core/test_model.py

```diff
@@ -290,6 +291,33 @@ def test_model_download(tmpdir, cls, filename):
         assert isinstance(task, cls)
 
 
+class DummyTask(Task):
+    def __init__(self):
+        super().__init__()
+
+        self.model = nn.Sequential(
+            nn.Linear(10, 20),
+            nn.Linear(20, 30),
+            nn.Linear(30, 40),
+        )
+
+    def predict_step(self, batch, batch_idx: int, dataloader_idx: int = 0):
+        return self.model(batch)
+
+
+def test_as_embedder():
+    layer_number = "1"
+    embedder = DummyTask().as_embedder(f"model.{layer_number}")
```

Comment:
just a question: Should we also allow to only call `.as_embedder(layer_number)`? I.e. splitting after the last separator '.' and check those as well? I think almost all builtin tasks only have a single model, right?

Reply:
> Or at least remove the initial attribute name within the task.

Definitely, I think there are a few enhancements here that would make it easier for users to know which layer to choose for embedding. It would also be cute if we could add a `backbone` layer or something that just goes through everything except the model head, but we may get that for free due to our naming conventions.

I'll leave these enhancements out of this PR as I need to play around with it a bit to figure out what the challenges are, but definitely some things we should look in to!